### ISSUE TITLE
Add Scene management

### ISSUE THREAD

**Issue by straker** [straker/kontra#122]
Had a few requests to add this. See #115 for some example implementations.

**Comment by BoldBigflank** [comment]
I'm thinking through this and if a gameLoop becomes basically a scene, the gameLoop itself should probably have a collection of sprites that it manages, and it might need a very simple update and render function to start:
`this.sprites.forEach(s => s.update())`
`this.sprites.forEach(s => s.render())`
Then it would need an addSprite function that pushes that sprite to its array.

Then have a small update to Sprite to pass it in on initialization:
```
  constructor(properties, loop) {
    this.init(properties);
    if (loop) loop.addSprite(this)
  }
```
So you would instantiate a sprite, then say what gameLoop it belongs to:
```
var s = new Sprite(foo, loop)
```

Are there issues with this that I haven't thought of?

The gameLoop should also be able to change whether it clears on every frame (because stacking multiple would cause issues). It would also be nice to have some other states along with start and stop: pause and hidden (or background, idk what to call it). Pause would do the render functions each frame but not the update, and hidden would do the update function for all its sprites and not the render function.

A gameLoopManager would have the stack of gameLoops and manage which one clears after every frame.

**Comment by straker** [comment]
Interesting idea. I'd be concerned about auto-rendering sprites since order might be important. We could get super fancy and do auto-sprite ordering as in #94, but I'm still unsure how complex we would want to get.

I'm also not sure I want to add a scene (or loop) as an argument of sprite. Ties them together too much.

**Comment by straker** [comment]
Closed via #169
